### DISCUSSION TITLE
Merge queues

### DISCUSSION THREAD

**Discussion by Alizter** [HoTT/HoTT#1749] [General]
I've enabled merge queues in the repo meaning that rather than pressing merge, you have to "add to merge queue". PRs in the queue will be sequentially CI'd and merged if successful. This way we don't have to wait for a green CI if we are happy with a PR being merged.

**Comment by jdchristensen** [comment]
This seems to force two of the CI checks to run again, even if they already successfully passed.  I guess this is in case a commit that happened in the meantime affects the success of the current PR.  But for a project that doesn't get a huge amount of traffic, it might be better to be more lightweight.  Is it possible to add an option to bypass the queue?  Or maybe it's not really needed for this project?

**Reply by jdchristensen** [reply]
In the test I just did, I had to wait 6-7 minutes for this before I could update my local copy of the master branch.

**Reply by Alizter** [reply]
It didn't work out the way I had hoped so disabling this.

**Reply by jdchristensen** [reply]
Is it possible to make it optional?  I think it could be useful in many cases, e.g. if I push a final fix and then am going to be busy with something else, I could just hit that button.  But sometimes we want to be able to merge instantly.

**Reply by Alizter** [reply]
There doesn't appear to be a way to make it optional. It's part of the "branch protection rules" which means I can only require it always or never. It's a shame but I guess we don't have so much traffic so it would never be useful enough I suppose.
